pdns/lock.hh

```diff
@@ -288,7 +288,7 @@ public:
     return LockGuardedHolder<T>(d_value, d_mutex);
   }
 
-  LockGuardedHolder<const T> read_only_lock() const
```

Comment:
Would it make sense to have both a `const` and non-`const` variant here? e.g.
```
  LockGuardedHolder<const T> read_only_lock() const
  {
    return LockGuardedHolder<const T>(d_value, d_mutex);
  }

  LockGuardedHolder<T> read_only_lock()
  {
    return LockGuardedHolder<T>(d_value, d_mutex);
  }
```

Reply:
Hmm, why would we want a writable version of the data with a read-only lock?

Reply:
Hmm, ok, I was confused. Of course the method should not be `const`, and it does correcly return a `const T` holder.